bridges/DealabsBridge.php

```diff
@@ -2081,6 +2101,128 @@ protected function collectDeals($url){
 		}
 	}
 
+	/**
+	 * Get the Talk lastest comments
+	 */
+	protected function collectDataTalk(){
+		$threadURL = $this->getInput('url');
+		$onlyWithUrl = $this->getInput('only_with_url');
+
+		// Get Thread ID from url passed in parameter
+		$threadSearch = preg_match('/-([0-9]{1,20})$/', $threadURL, $matches);
+
+		// Show an error message if we can't find the thread ID in the URL sent by the user
+		if($threadSearch !== 1) {
+			returnClientError($this->i8n('thread-error'));
+		}
+		$threadID = $matches[1];
+
+		$url = $this->i8n('bridge-uri') . 'graphql';
+
+		// Get Cookies header to do the query
+		$cookies = $this->getCookies($url);
+
+		// Construct the GraphQL query string
+		$graphqlArray = array(
+			'query comments($filter:CommentFilter!,$limit:Int,$page:Int){comments(filter:$filter,limit:$li',
+			'mit,page:$page){items{...commentFields}pagination{...paginationFields}}}fragment commentField',
+			's on Comment{commentId threadId url preparedHtmlContent user{...userMediumAvatarFields...user',
+			'NameFields...userPersonaFields bestBadge{...badgeFields}}reactionCounts{type count}deletable ',
+			'currentUserReaction{type}reported reportable source status createdAt updatedAt ignored popula',
+			'r deletedBy{username}notes{content createdAt user{username}}lastEdit{reason timeAgo userId}}f',
+			'ragment userMediumAvatarFields on User{userId isDeletedOrPendingDeletion imageUrls(slot:"defa',
+			'ult",variations:["user_small_avatar"])}fragment userNameFields on User{userId username isUser',
+			'ProfileHidden isDeletedOrPendingDeletion}fragment userPersonaFields on User{persona{type text',
+			'}}fragment badgeFields on Badge{badgeId level{...badgeLevelFields}}fragment badgeLevelFields ',
+			'on BadgeLevel{key name description}fragment paginationFields on Pagination{count current last',
+			' next previous size order}'
+		);
+		$graphqlString = implode('', $graphqlArray);
```

Comment:
How about using multline php string?

Reply:
Hmm, do you mean using the Heredoc syntax :
https://www.php.net/manual/en/language.types.string.php#language.types.string.syntax.heredoc
?

Cutting this string to words is a bit of nightmare, but I can do it of needed !

Reply:
> do you mean using the Heredoc syntax :

Yes, I mean using heredoc syntax.

> Cutting this string to words is a bit of nightmare, but I can do it of needed !

The nightmare is reading this graphql query.

As an alternative, if you copied that graphql query from some code as is, you could just point where to get this string.

Reply:
Ok, I understand :)
I didn't write this GraphQL Query : I discovered the name while writing the bridge !

This query was extracted using the network tab of Firefox while visiting the website !

I used nowdoc (because some content looked like PHP in it.